src/python/aqueduct_executor/operators/connectors/data/s3.py

```diff
@@ -127,7 +127,16 @@ def extract(self, params: extract.S3Params) -> Any:
             if len(path) == 0:
                 raise Exception("S3 file path cannot be an empty string.")
             if path[-1] == "/":
-                # This means the path is a directory, and we will do a prefix search.
+                # Check whether the folder path exists in S3 by just checking the folder
+                # rather than the content in the folder. `MaxKeys = 2` allows file and
+                # directory sharing the same name. The return value will only contains
+                # `CommonPrefixes` if there is a folder match.
```

Comment:
@Fanjia-Yan I'm a bit confused by the comment here. What does "`MaxKeys = 2` allows file and directory sharing the same name" mean? Also what does `CommonPrefixes` mean? Do you have a pointer to the documentation that I can take a look at?

Reply:
Here is the doc for the API https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/list_objects.html. So what I am trying to do here is to distinguish empty file folder and non-existing file folder. To do that, say I'm querying on folder `aqueduct/` I'm searching over the its parent directory to see `aqueduct` exist and adding the `delimitter`, the output in `CommonPrefixes`will have `\` which indicates that the folder exists.

Reply:
Yeah I'm also pretty confused. What do you mean searching over the parent directly? Aren't we just querying the folder directly, as you said in the commend above?

Reply:
I think I know what you mean. Sorry for the confusion, let me try if I can get a cleaner way to do this.